### PR TITLE
Implement __dir__ on _ContextMap

### PR DESCRIPTION
**Pull request by ekohl**
This allows for easy introspection which ipython can gladly use.

### CODE REVIEW

File: octomachinery/app/runtime/utils.py [hunk 1]
```diff
@@ -19,6 +19,9 @@ def __init__(self, **initial_vars):
         self.__token_map__ = {}
         """Storage for individual context var reset tokens."""
 
+    def __dir__(self):
```
Comment (webknjaz) [inline octomachinery/app/runtime/utils.py]:
```suggestion
    def __dir__(self):
        """Render a list of public attributes."""
```

**Comment by webknjaz** [comment]
@ekohl it's now in v0.0.13

**Comment by webknjaz** [comment]
Well, will be once https://travis-ci.com/sanitizers/octomachinery/jobs/190630840 is finished.

**Comment by webknjaz** [comment]
Done: https://pypi.org/project/octomachinery/0.0.13/

**Comment by ekohl** [comment]
I was just toying with it in ipython to explore the API. I had hoped to convince [jedi-vim](https://github.com/davidhalter/jedi-vim) to do something more useful on `RUNTIME_CONTEXT` completion but that might be a bit too dynamic. This was already a good start :)

**Comment by webknjaz** [comment]
Yes, it's too dynamic. pylint also fails to recognize attrs which are not in slots.

**Comment by webknjaz** [comment]
For now, all APIs you need are in the tutorial and in the README example :)

**Comment by ekohl** [comment]
My initial search was finding out the API specification of the github client. How to translate the GH API docs into actual API calls would be useful. Items I struggled with:
* My goal was to set labels. Can I omit the body?
* How do I define the parameters to a function with `@process_webhook_payload`? Do I need to list all parameters that the webhook sends or can I list only those I'm interested in?

**Comment by webknjaz** [comment]
API client loosely wraps what https://gidgethub.readthedocs.io provides (https://github.com/sanitizers/octomachinery/blob/master/octomachinery/github/api/raw_client.py#L101-L106) adding `preview_api_version` param to be used for preview APIs. With it you don't have to specify the whole Accept header value but only its unique part.

Also, this inherits (for now) Gidgethub's decision to always require a body but you can provide an empty body using `b''` as a workaround.

If you use `@process_webhook_payload` it just blindly unpacks all keys from the webhook's JSON payload without attempting to inspect the coroutine signature. This may change in the future.
For now, you should either list all of the args (some of them will require a default, i.e. `arg=None` because they are optional for different event actions). You can go to your GitHub App page (where you downloaded the pem key) and then go to `Advanced` menu item to see what GitHub has sent to your App and inspect what the actual payload was to get an idea of what args you may need.
Another way would be to list args you need and then just put `**kwargs` catch-all.

**Comment by webknjaz** [comment]
@ekohl oh and you can omit the decorator to just get an event object and get the data from enclosed dict manually.
Example: https://github.com/sanitizers/chronographer-github-app/blob/f1eb1ca/chronographer/event_handlers.py#L69-L100

**Pull request merged**